lib/autoupdate/agent/updater.go

```diff
@@ -67,6 +71,17 @@ const (
 	errorKey  = "error"
 )
 
+// SetRequiredUmask sets the umask to match the systemd umask that the teleport-update service will execute with.
+// This ensures consistent file permissions.
+// NOTE: This must be run in main.go before any goroutines that create files are started.
+func SetRequiredUmask(ctx context.Context, log *slog.Logger) {
```

Comment:
Can you add some tests around this and the updater itself to catch regressions?

Reply:
I plan to add a case to @hugoShaka's Ansible integration test suite to ensure it's set appropriate throughout the whole workflow: https://github.com/gravitational/teleport.e/tree/hugo/autoupdate-ansible-playbook/tests/ansible-autoupdate-playbook (private link)

I'll add a unit test to the bitwise conditional on the warning. I'm hesitant to call `syscall.Umask` in unit tests, since it impacts all tests in parallel and could lead to very hard to debug flakes.